### PR TITLE
Move "edit this page" to top of page

### LINKED ISSUE #1359: Should we move "edit this page" link to top of the page?
Putting it at the end is kinda useless.

### PR DESCRIPTION
**Pull request by willium**
Fixes #1359

It'd be great if you could run it and try it out. On second thought, stylistically, I'm not sure I like this.

Reasoning:
- It's not clear where to put it (especially on the homepage, and smaller screens (though I suppose it could be hidden on smaller screens)
- 99% of readers don't/won't/shouldn't edit pages, and it's given a lot of prominence
- it doesn't make good use of space (especially on homepage)
- people generally look for this link in the footer (citation needed)
- Why is putting it at the end useless? Have people had trouble finding it?

### CODE REVIEW

**Comment by domoritz** [comment]
I feel similar things. Maybe it makes sense to only do it in the docs (have two links, at the top and bottom). I'd even be okay with a small link right below the title.

**Comment by willium** [comment]
Thoughts @kanitw?

**Comment by kanitw** [comment]
I agree that we shouldn't do this on the homepage and for small screens.   

> people generally look for this link in the footer (citation needed)

> Why is putting it at the end useless? Have people had trouble finding it?

I am pretty sure that 90% of people don't actively look for this link.  
If we don't put the link on the top, the problem is they don't even know that they can edit the docs.  

> 99% of readers don't/won't/shouldn't edit pages, and it's given a lot of prominence

Maybe this will change the number to 98%?   At worst, they will just scroll through it and do nothing.  At least they'll notice that they can contribute.

**Comment by domoritz** [comment]
Okay, then that's decided. Let's put a second edit link at the top of the docs but no other page.

**Comment by willium** [comment]
Should we put it in the footer on other pages?

**Comment by domoritz** [comment]
Just leave it where it is on all pages but add a link on the docs pages.

**Comment by willium** [comment]
Me gusta. will do tonight.

**Comment by domoritz** [comment]
Tonight ten days ago :-P

**Comment by willium** [comment]
err.... yeah I'll get on that 😬

**Comment by domoritz** [comment]
ping

**Comment by willium** [comment]
Should finally be ready... @domoritz @kanitw

File: site/static/main.css [hunk 1]
```diff
@@ -277,10 +278,31 @@ nav a:hover {
   flex: 0 0 auto;
 }
 
+#edit-this-page {
+  display: inline-block;
+  font-size: 0.9em;
+  color: #aaa;
+  padding-top: 30px;
+  cursor: pointer;
```
Comment (domoritz) [inline site/static/main.css]: This is weird. It feel like I can click on the whole text but it's not possible.

File: _layouts/docs.html [hunk 2]
```diff
@@ -40,5 +40,5 @@
 
 base: /docs/
 ---
-
+{% include edit.html %}
```
Comment (domoritz) [inline _layouts/docs.html]: I'd prefer a shorter version at the top that does not have the "and submit a pull request!"

**Comment by domoritz** [comment]
LGTM except for the two comments. Will merge after they are done.

**Comment by willium** [comment]
@domoritz ready

File: _layouts/docs.html [hunk 3]
```diff
@@ -40,5 +40,7 @@
 
 base: /docs/
 ---
-
+<span id="edit-this-page">
```
Comment (domoritz) [inline _layouts/docs.html]: Doesn't `edit-this-page` have to be a class now?

Comment (willium) [inline _layouts/docs.html]: Sigh. Yes

Comment (domoritz) [inline _layouts/docs.html]: Sorry for being so pedantic.

Comment (willium) [inline _layouts/docs.html]: No no you're right. I'll fix soon

**Pull request merged**